### PR TITLE
added polygon directive

### PR DESCRIPTION
**Pull request by allenhwkim**
Here are few things to notice

1. `ngc` did not work for me to copy files to dist directory. Instead it asked me a `name`. Thus, I had to manually run the following to build instead of just running npm run build

  1) tsc
  2) npm run build:umd
  3) npm run build:app

  NOTE: this issue should be resolved and clearly written in README.md when we find a solution

2. I noticed the following directories are generated for all `.d.ts`, which was not there and unnecessary 

    1) dist/dist
    2) app/build/dist

  NOTE: this issue should be resolved  in the future

3) .metadata.json has been deleted. I do not know how to generate these files. README.md should be updated to generate these files.

  NOTE:  README.md  should be update fo how to generate these files

4) Two files are added for debugging purpose, and find that somewhat webpack does not give me enough information to debug.

    1) debug.html
    2) systemjs.config.js

### CODE REVIEW

**Comment by allenhwkim** [comment]
Here are few things to notice

1. ngc did not work, so I had to manually run the following to build.

  1) tsc
  2) npm run build:umd
  3) npm run build:app

2. I noticed the following directories are generated for all `.d.ts`, which was not there and unnecessary 

    1) dist/dist
    2) app/build/dist


3) .metadata.json has been deleted. I do not know how to generate these files. README.md should be updated to generate these files.

File: src/services/option-builder.ts [hunk 1]
```diff
@@ -39,39 +39,42 @@ export class OptionBuilder {
...
-        /*  MapTypeID.HYBRID -> new google.maps.MapTypeID.HYBRID */
-        || this.getAnyMapConstant(input, options)
-
-        /*  2016-06-20 -> new Date('2016-06-20') */
-        || this.getDateObject(input)
-
-        || input;
+    let output: any = input;
+    if (typeof input === 'string') { //convert string to a google object
+      if (input === 'false') {
```
Comment (aitboudad) [inline src/services/option-builder.ts]: should we really handle those cases (`false` and `0`) ?

Comment (allenhwkim) [inline src/services/option-builder.ts]: It's from ngMap(Angular1)  experience. It was a bug fix.

**Comment by aitboudad** [comment]
1. `ngc` you should use `npm run build:ngc` 
2. `.metadata.json` are  generated by `ngc`
4. for webpack https://github.com/ng2-ui/ng2-map/issues/31#issuecomment-260217536

**Comment by allenhwkim** [comment]
`ngc` or  `npm run build:ngc` it did not work as I expected. It does not copy source code. Could you double-check `ngc` copies all source code?

**Comment by allenhwkim** [comment]
`npm run build:ngc` or `ngc` asks me a name, which interrupts the building process, and it creates a directory, which is very different result from `tsc`

What am I missing?

image

```
ng2-map$ (29-polygon) rm -rf ng2-map
ng2-map$ (29-polygon) rm -rf ngc
ng2-map$ (29-polygon) rm -rf dist
ng2-map$ (29-polygon) npm run build:ngc

> ng2-map@0.7.0 build:ngc /Users/allen.kim/github/ng2-map
> ngc

Name: dist
Component created
ng2-map$ (29-polygon) git sttus
git: 'sttus' is not a git command. See 'git --help'.

Did you mean this?
        status
ng2-map$ (29-polygon) git status
On branch 29-polygon
Your branch is up-to-date with 'origin/29-polygon'.
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        deleted:    dist/components/info-window.d.ts
        deleted:    dist/components/info-window.js
        deleted:    dist/components/info-window.js.map
        deleted:    dist/components/ng2-map.component.d.ts
        deleted:    dist/components/ng2-map.component.js
        deleted:    dist/components/ng2-map.component.js.map
        deleted:    dist/directives/circle.d.ts
        deleted:    dist/directives/circle.js
        deleted:    dist/directives/circle.js.map
        deleted:    dist/directives/marker.d.ts
        deleted:    dist/directives/marker.js
        deleted:    dist/directives/marker.js.map
        deleted:    dist/directives/polygon.d.ts
        deleted:    dist/directives/polygon.js
        deleted:    dist/directives/polygon.js.map
        deleted:    dist/dist/components/info-window.d.ts
        deleted:    dist/dist/components/ng2-map.component.d.ts
        deleted:    dist/dist/directives/circle.d.ts
        deleted:    dist/dist/directives/marker.d.ts
        deleted:    dist/dist/directives/polygon.d.ts
        deleted:    dist/dist/index.d.ts
        deleted:    dist/dist/services/geo-coder.d.ts
        deleted:    dist/dist/services/navigator-geolocation.d.ts
        deleted:    dist/dist/services/ng2-map.d.ts
        deleted:    dist/dist/services/option-builder.d.ts
        deleted:    dist/dist/services/util.d.ts
        deleted:    dist/index.d.ts
        deleted:    dist/index.js
        deleted:    dist/index.js.map
        deleted:    dist/ng2-map.umd.js
        deleted:    dist/ng2-map.umd.js.map
        deleted:    dist/services/geo-coder.d.ts
        deleted:    dist/services/geo-coder.js
        deleted:    dist/services/geo-coder.js.map
        deleted:    dist/services/navigator-geolocation.d.ts
        deleted:    dist/services/navigator-geolocation.js
        deleted:    dist/services/navigator-geolocation.js.map
        deleted:    dist/services/ng2-map.d.ts
        deleted:    dist/services/ng2-map.js
        deleted:    dist/services/ng2-map.js.map
        deleted:    dist/services/option-builder.d.ts
        deleted:    dist/services/option-builder.js
        deleted:    dist/services/option-builder.js.map
        deleted:    dist/services/util.d.ts
        deleted:    dist/services/util.js
        deleted:    dist/services/util.js.map

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        dist/dist.component.js
        dist/dist.html
        dist/dist.scss

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by aitboudad** [comment]
It didn't happen to me :) anyway I'll try to find what's going wrong

**Pull request merged**